### ISSUE TITLE
inspect.lua doesn't work

### ISSUE THREAD

**Issue by chobie** [yuin/gopher-lua#10]
Hi, I've just played with inspect.lua (https://github.com/kikito/inspect.lua) but it seems it doesn't work correctly.

for example)
```
macbook% ../bin/glua bug.lua

inspect.lua:300: attempt to index a non-table object(nil)
stack traceback:
inspect.lua:300: in inspect.lua:297
(tailcall): ?
bug.lua:2: in main chunk
[G]: ?
macbook% lua bug.lua
"hello"
```

https://gist.github.com/chobie/bcff5c74c915422a3075

also, I've tested above script with lua 5.1.5 and lua 5.2.3 and it works fine.
can you check this if you have a chance?

Thanks,

**Comment by chobie** [comment]
minimal reproduce code is here.

```
function inspect(options)
	options       = options or {}
	print(type(options))
	print(options.depth)
end

inspect(nil)
```

```
# glua
macbook% ../bin/glua spec.lua
nil

spec.lua:4: attempt to index a non-table object(nil)
stack traceback:
	spec.lua:4: in inspect
	spec.lua:7: in main chunk
	[G]: ?

#lua-5.1
macbook% lua-5.1 spec.lua
table
nil
```

**Comment by yuin** [comment]
@chobie , Thank you for reporting this.
I've fixed the bug in commit ca03869a34d49d421da32815aec6920b9ee30b7f.
Could you test it?

**Comment by chobie** [comment]
Thanks. it seems good for above problem. but also I've found another issue related this.

```
function inspect(options)
	a = options or setmetatable({}, {__mode = "test"})
	print(type(a))
end

function inspect2(options)
	options = options or setmetatable({}, {__mode = "test"})
	print(type(options))
end

inspect(nil)
inspect2(nil)

#
spec.lua:7: attempt to call a non-function object
stack traceback:
	spec.lua:7: in inspect2
	spec.lua:12: in main chunk
	[G]: ?
```

**Comment by yuin** [comment]
Hmm, I've fixed it again in fd7274ce01679887249cfd78047832230ba98052.
Could you test it again?

**Comment by chobie** [comment]
Thanks, but more problems appeared.

* [G]: error parsing regexp: trailing backslash at end of expression: ``

```
str = "hello"
str:gsub("\\", "\\\\")
```

I guess `inspect.lua` will work fine when this problem solved.

**Comment by yuin** [comment]
Oh, I've fixed it in 67b4bef33ee0e4fc01320b00a372101c908c552c.
Could you test it once again?

**Comment by chobie** [comment]
Thank you so much! It works fine :)
